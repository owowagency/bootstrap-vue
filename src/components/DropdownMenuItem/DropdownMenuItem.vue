<template>
    <li>
        <hr
            v-if="divider"
            class="dropdown-divider"
            :class="itemClass"
        >

        <h6
            v-else-if="header"
            class="dropdown-header"
            :class="itemClass"
        >
            <!-- @slot Displays content inside the dropdown menu item. -->
            <slot name="default">
                {{ label }}
            </slot>
        </h6>

        <Component
            :is="is"
            v-else
            class="dropdown-item"
            :class="[{active, disabled}, itemClass]"
            :href="href"
            :to="to"
            @click.prevent
        >
            <slot name="default">
                {{ label }}
            </slot>
        </Component>
    </li>
</template>

<script lang="ts" setup>
import {PropType, computed} from 'vue';
import {dropdownItemProps} from '@/composables/useDropdownItem';

const props = defineProps({
    /**
     * Adds the `active` class
     */
    active: {
        type: Boolean,
        default: false,
    },
    /**
     * Adds the `disabled` class
     */
    disabled: {
        type: Boolean,
        default: false,
    },
    /**
     * Displays a divider
     */
    divider: {
        type: Boolean,
        default: false,
    },
    ...dropdownItemProps,
    /**
     * Displays label as header
     */
    header: {
        type: Boolean,
        default: false,
    },
    href: {
        type: String,
        default: '#',
    },
    is: {
        type: String,
        default: undefined,
    },
    /**
     * The label to display
     */
    label: {
        type: String,
        default: undefined,
    },
    to: {
        type: [Object, String] as PropType<string|Record<string, unknown>>,
        default: undefined,
    },
});

const is = computed(() => {
    if (props.is) {
        return props.is;
    }

    if (props.to) {
        return 'RouterLink';
    }

    return 'a';
});
</script>

<docs>
```vue
<DropdownMenuItem label="Menu item" />
```
</docs>
