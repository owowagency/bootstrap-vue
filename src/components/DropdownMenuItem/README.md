# DropdownMenuItem

```vue
<DropdownMenuItem label="Menu item" />
```

## Props

| Prop name | Description               | Type                                  | Values | Default   |
| --------- | ------------------------- | ------------------------------------- | ------ | --------- |
| active    | Adds the `active` class   | boolean                               | -      | false     |
| disabled  | Adds the `disabled` class | boolean                               | -      | false     |
| divider   | Displays a divider        | boolean                               | -      | false     |
| header    | Displays label as header  | boolean                               | -      | false     |
| href      |                           | string                                | -      | '#'       |
| is        |                           | string                                | -      | undefined |
| label     | The label to display      | string                                | -      | undefined |
| to        |                           | string\|Record&lt;string, unknown&gt; | -      | undefined |

## Slots

| Name    | Description                                     | Bindings |
| ------- | ----------------------------------------------- | -------- |
| default | Displays content inside the dropdown menu item. |          |
