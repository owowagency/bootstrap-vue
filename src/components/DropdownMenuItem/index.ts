import DropdownMenuItem from './DropdownMenuItem.vue';

export default DropdownMenuItem;
