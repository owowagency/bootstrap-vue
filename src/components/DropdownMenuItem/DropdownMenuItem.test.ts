import {describe, expect, it} from 'vitest';
import DropdownMenuItem from '@/components/DropdownMenuItem';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentSlotRenderTest(DropdownMenuItem);

    componentRenderTest(DropdownMenuItem, {props: {href: 'https://github.com/owowagency/bootstrap-vue'}});

    componentRenderTest(DropdownMenuItem, {props: {is: 'div'}});

    componentRenderTest(DropdownMenuItem, {props: {to: {name: 'home'}}});

    componentSlotRenderTest(
        DropdownMenuItem,
        'default',
        {props: {header: true}},
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {label: 'Hello'}},
        true,
        'renders label',
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {active: true, label: 'there'}},
        true,
        'renders active',
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {disabled: true, label: 'there'}},
        true,
        'renders disabled',
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {divider: true}},
        true,
        'renders divider',
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {header: true, label: 'general kenobi'}},
        true,
        'renders header',
    );

    it('adds item class to link', () => {
        const wrapper = shallowMount(DropdownMenuItem, {
            props: {
                itemClass: ['hello-there'],
            },
        });

        expect(wrapper.find('a').classes()).toContain('hello-there');
    });
});
