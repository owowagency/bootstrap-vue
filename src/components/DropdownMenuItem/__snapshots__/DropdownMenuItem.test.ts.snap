// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders active 1`] = `"<li><a class=\\"dropdown-item active\\" href=\\"#\\">there</a></li>"`;

exports[`template > renders default 1`] = `"<li><a class=\\"dropdown-item\\" href=\\"https://github.com/owowagency/bootstrap-vue\\"></a></li>"`;

exports[`template > renders default 2`] = `
"<li>
  <div class=\\"dropdown-item\\" href=\\"#\\"></div>
</li>"
`;

exports[`template > renders default 3`] = `
"<li>
  <routerlink class=\\"dropdown-item\\" href=\\"#\\" to=\\"[object Object]\\"></routerlink>
</li>"
`;

exports[`template > renders disabled 1`] = `"<li><a class=\\"dropdown-item disabled\\" href=\\"#\\">there</a></li>"`;

exports[`template > renders divider 1`] = `
"<li>
  <hr class=\\"dropdown-divider\\">
</li>"
`;

exports[`template > renders header 1`] = `
"<li>
  <h6 class=\\"dropdown-header\\">
    <!-- @slot Displays content inside the dropdown menu item. -->general kenobi
  </h6>
</li>"
`;

exports[`template > renders label 1`] = `"<li><a class=\\"dropdown-item\\" href=\\"#\\">Hello</a></li>"`;
